dnf/comps.py

```diff
@@ -395,21 +395,83 @@ def groups_iter(self):
         return (self._build_group(g) for g in self._i.groups)
 
 
+class CompsTransPkg(object):
+    def __init__(self, pkg_or_name):
+        if dnf.util.is_string_type(pkg_or_name):
+            self.basearchonly = False
+            self.name = pkg_or_name
+            self.optional = True
+            self.requires = None
+        else:
+            self.basearchonly = pkg_or_name.basearchonly
+            self.name = pkg_or_name.name
+            self.optional = pkg_or_name.type & libcomps.PACKAGE_TYPE_OPTIONAL
+            self.requires = pkg_or_name.requires
+
+    def __eq__(self, other):
+        return (self.name == other.name and
+                self.basearchonly == self.basearchonly and
+                self.optional == self.optional and
+                self.requires == self.requires)
+
+    def __str__(self):
+        return self.name
+
+    def __hash__(self):
+        return hash((self.name,
+                    self.basearchonly,
+                    self.optional,
+                    self.requires))
+
+
 class TransactionBunch(object):
     def __init__(self):
-        self.install = set()
-        self.install_opt = set()
```

Comment:
@mluscon If again we will add strict behavior for mandatory packages, there is no easy switch how to enable it? Or did I miss something?